### ISSUE TITLE
servo panics on exit on windows 10 x64

### ISSUE THREAD

**Issue by yoava333** [servo/servo#10547]
I'm using ea97c8905580e67e96ae00572d1c4e7af3d58736 windows 10 x64 (build 10586.104)

```
$ ./mach run -r "https://github.com"
ERROR:js::rust: Error at https://assets-cdn.github.com/assets/frameworks-17fc21583b56cf4c511436a15e7eb5f4f48d3d8c1006e654d3108ef67a707cae.js:6:23659: history is not defined

thread '<main>' panicked at 'called `Option::unwrap()` on a `None` value', ../src/libcore\option.rs:326
stack backtrace:
   0:          0x1f5255a - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::hff309ab1d83ffd90
   1:          0x1f51a7e - std::panicking::default_hook::h08ad3bb09872855b
   2:          0x1f14383 - std::sys_common::unwind::begin_unwind_inner::h406d5f1a330b854b
   3:          0x1f15f1b - std::sys_common::unwind::begin_unwind_fmt::h57ea3fbee1a40196
   4:          0x1f44bab - rust_begin_unwind
   5:          0x1f66ca5 - core::panicking::panic_fmt::ha6b3c19493c123b3
   6:          0x1f677b0 - core::panicking::panic::h188c6c6a0fe5463c
   7:           0xfd5456 - glutin_app::window::Window::wait_events::h1f6001f3a4f44ef0
   8:           0x40341a - servo::main::h6924e5658370f6b5
   9:          0x1f5109d - std::sys_common::unwind::try::hb63b2f006f75d989
  10:          0x1f50eca - std::rt::lang_start::h12d///9b3310b48a726
  11:           0x4013b4 - _tmainCRTStartup
  12:           0x4014e7 - mainCRTStartup
  13:     0x7ffd86108101 - mz_zip_writer_add_to_central_dir.constprop.15
Servo exited with return value 101
```

**Comment by asajeffrey** [comment]
Looks like the culprit is the `unwrap()` in `glutin/window.rs`:
```rust
    handle_next_event(&self) -> bool {
        let event = self.window.wait_events().next().unwrap();
```

**Comment by asajeffrey** [comment]
There are a lot of uses of unwrap in glutin, which should be removed, and we should add glutin/* to the unwrap-is-banned list.

**Comment by yoava333** [comment]
I've looked into this panic. It's caused by 
```
let event = self.window.wait_events().next().unwrap();
```
the reason that we get None is that on Windows, is that glutin receives events on a channel and the channel is closed when we try to read from it.
The transmitting end of the channel is dropped when the event loop thread shuts down (https://github.com/servo/glutin/blob/servo/src/api/win32/init.rs#L80). 
this is due to the thread's message queue closing because I closed the window.

On my windows machine, the Event::Closed is sometimes received before the panic, and sometimes not. either way we get a panic because the channel is closed before we get to ShutdownState::FinishedShuttingDown state.
